lib/facade_converter.ts

```diff
@@ -147,6 +118,58 @@ export class FacadeConverter extends base.TranspilerBase {
     this.emit(ident);
   }
 
+  shouldEmitNew(c: ts.CallExpression): boolean {
+    if (!this.tc) return true;
+
+    let {context, symbol} = this.getCallInformation(c);
+    if (!symbol) {
+      // getCallInformation returns a symbol if we understand this call.
+      return true;
+    }
+
+    var loc = this.getFileAndName(c, symbol);
+    if (!loc) return true;
+    var {fileName, qname} = loc;
+    var fileSubs = this.callHandlerReplaceNew[fileName];
+    if (!fileSubs) return true;
+    return !fileSubs[qname];
+  }
+
+  private getCallInformation(c: ts.CallExpression): {context?: ts.Expression, symbol?: ts.Symbol} {
+    var symbol: ts.Symbol;
+    var context: ts.Expression;
+    var ident: string;
+
+    if (c.expression.kind === ts.SyntaxKind.Identifier) {
+      // Function call.
+      ident = base.ident(c.expression);
+      if (!this.candidateProperties.hasOwnProperty(ident)) return {};
```

Comment:
This block still duplicates a lot of code from the `if` block immediately following it. Can't you assign into `ident` and `symbol` in the if statements, and then execute the same code as below, potentially with a check when to return an error vs an empty object?